### DISCUSSION TITLE
Id generation in migrations

### DISCUSSION THREAD

**Discussion by bmingles** [pocketbase/pocketbase#3912] [Q&A] [answered]
Is the api that creates ids for collections, fields, indexes, etc. accessible inside of migrations? I'm writing some utils for my migrations and would like to be able to do things like:

```
new Collection({
  id: uniqueId(),
  ...
})
```

I'd prefer being able to use the same function Pocketbase uses if possible.

**Comment by ganigeorgiev** [comment, marked as answer]
The ids are autogenerated if not expliticly set, but for the models you can invoke the [`refreshId()` method](https://pocketbase.io/jsvm/classes/Collection.html#refreshId) if you want to generate the id before save:

```js
const col = new Collection();
col.refreshId();
console.log(col.id); // should be random generated
```

Alternatively, you can use the random string helpers available under the [`$secuirty.*` namespace](https://pocketbase.io/jsvm/modules/_security.html).

**Reply by bmingles** [reply]
Great, where can I find the source code for the `refreshId` function?

**Reply by bmingles** [reply]
I think I found it here: https://github.com/pocketbase/pocketbase/blob/985ab1e5b79a328b516decca3d1d7dc8edfcbba0/models/base.go#L102
